qiskit/circuit/add_control.py

```diff
@@ -239,31 +239,46 @@ def apply_basic_controlled_gate(circuit, gate, controls, target):
                 circuit.cu(theta, phi, lamb, 0, controls[0], target)
         else:
             if phi == -pi / 2 and lamb == pi / 2:
-                circuit.mcrx(theta, controls, target, use_basis_gates=True)
+                circuit.mcrx(theta, controls, target, use_basis_gates=False)
             elif phi == 0 and lamb == 0:
                 circuit.mcry(
                     theta,
                     controls,
                     target,
-                    use_basis_gates=True,
+                    use_basis_gates=False,
                 )
             elif theta == 0 and phi == 0:
                 circuit.mcp(lamb, controls, target)
             else:
-                circuit.mcp(lamb, controls, target)
-                circuit.mcry(
-                    theta,
-                    controls,
-                    target,
-                    use_basis_gates=True,
-                )
-                circuit.mcp(phi, controls, target)
+                circuit.mcrz(lamb, controls, target, use_basis_gates=False)
+                circuit.mcry(theta, controls, target, use_basis_gates=False)
+                circuit.mcrz(phi, controls, target, use_basis_gates=False)
+                circuit.mcp((phi + lamb) / 2, controls[1:], controls[0])
```

Comment:
Nice optimization :).

Reply:
done in [4823321](https://github.com/Qiskit/qiskit/pull/13801/commits/48233216b4af1649159b4dbf51d549e99be54e8f)